File: frontend/src/app/shared/model/LivroCategoria.ts
export enum LivroCategoria {
  ROMANCE = 'ROMANCE',
  FANTASIA = 'FANTASIA',
  FICCAO = 'FICCAO',
  AVENTURA = 'AVENTURA',
  MISTERIO = 'MISTERIO',
  TERROR = 'TERROR',
  BIOGRAFIA = 'BIOGRAFIA',
  HISTORIA = 'HISTORIA',
  FILOSOFIA = 'FILOSOFIA',
  CIENCIA = 'CIENCIA',
  SOCIOLOGIA = 'SOCIOLOGIA',
  LITERATURA_BRASILEIRA = 'LITERATURA_BRASILEIRA',
  LITERATURA_ESTRANGEIRA = 'LITERATURA_ESTRANGEIRA',
  POESIA = 'POESIA',
  FICCAO_JUVENIL = 'FICCAO_JUVENIL',
  DESCONHECIDA = 'DESCONHECIDA',
}

const categoriaMap = {
  [LivroCategoria.ROMANCE]: 'Romance',
  [LivroCategoria.FANTASIA]: 'Fantasia',
  [LivroCategoria.FICCAO]: 'Ficção',
  [LivroCategoria.AVENTURA]: 'Aventura',
  [LivroCategoria.MISTERIO]: 'Mistério',
  [LivroCategoria.TERROR]: 'Terror',
  [LivroCategoria.BIOGRAFIA]: 'Biografia',
  [LivroCategoria.HISTORIA]: 'História',
  [LivroCategoria.FILOSOFIA]: 'Filosofia',
  [LivroCategoria.CIENCIA]: 'Ciência',
  [LivroCategoria.SOCIOLOGIA]: 'Sociologia',
  [LivroCategoria.LITERATURA_BRASILEIRA]: 'Literatura Brasileira',
  [LivroCategoria.LITERATURA_ESTRANGEIRA]: 'Literatura Estrangeira',
  [LivroCategoria.POESIA]: 'Poesia',
  [LivroCategoria.FICCAO_JUVENIL]: 'Ficção juvenil',
  [LivroCategoria.DESCONHECIDA]: 'DESCONHECIDA',
}

export function getNomeCategoria(categoria: LivroCategoria): string {
  return categoriaMap[categoria] || LivroCategoria.DESCONHECIDA;
}

File: backend/src/main/java/com/gestaobiblioteca/domain/model/LivroCategoria.java
package com.gestaobiblioteca.domain.model;

import com.gestaobiblioteca.i18n.ApplicationResources;

/**
 * @author Guilherme
 */
public enum LivroCategoria {

	ROMANCE(ApplicationResources.getInstance().getMessage("livrocategoria.romance")),
	FANTASIA(ApplicationResources.getInstance().getMessage("livrocategoria.fantasia")),
	FICCAO(ApplicationResources.getInstance().getMessage("livrocategoria.ficcao")),
	AVENTURA(ApplicationResources.getInstance().getMessage("livrocategoria.aventura")),
	MISTERIO(ApplicationResources.getInstance().getMessage("livrocategoria.misterio")),
	TERROR(ApplicationResources.getInstance().getMessage("livrocategoria.terror")),
	BIOGRAFIA(ApplicationResources.getInstance().getMessage("livrocategoria.biografia")),
	HISTORIA(ApplicationResources.getInstance().getMessage("livrocategoria.historia")),
	FILOSOFIA(ApplicationResources.getInstance().getMessage("livrocategoria.filosofia")),
	CIENCIA(ApplicationResources.getInstance().getMessage("livrocategoria.ciencia")),
	SOCIOLOGIA(ApplicationResources.getInstance().getMessage("livrocategoria.sociologia")),
	LITERATURA_BRASILEIRA(ApplicationResources.getInstance().getMessage("livrocategoria.literatura_brasileira")),
	LITERATURA_ESTRANGEIRA(ApplicationResources.getInstance().getMessage("livrocategoria.literatura_estrangeira")),
	POESIA(ApplicationResources.getInstance().getMessage("livrocategoria.poesia")),
	FICCAO_JUVENIL(ApplicationResources.getInstance().getMessage("livrocategoria.ficcao_junvenil")),
	DESCONHECIDA(ApplicationResources.getInstance().getMessage("livrocategoria.desconhecida")),;

	LivroCategoria(String descricao) {
		this.descricao = descricao;
	}

	private final String descricao;

	public String getDescricao() {
		return descricao;
	}
}
